### ISSUE TITLE
test failure in Debian sid

### ISSUE THREAD

**Issue by henrich** [clear-code/cutter#27]
building cutter as cutter-test-framework packae in Debian, it gets test failure
Could you take a look at attached text, please?

[cutter.txt](https://github.com/clear-code/cutter/files/889998/cutter.txt)

**Comment by kou** [comment]
Thanks for your report. But it couldn't reproduce on my Debian GNU/Linux sid amd64.

Could you show your build environment?

It seems that the first one is caused by needless escape sequence.
The second one is occurred by memory destruction or invalid memory access.

**Comment by kenhys** [comment]
It seems that I could reproduce this situation.

```
% apt source cutter-testing-framework
% sudo DIST=sid pbuilder --build --distribution sid cutter-testing-framework_1.2.4-1.dsc 2>&1 | tee build.log    
```

**Comment by henrich** [comment]
Yes, I've tried it with 1.2.5 as below

```
$ apt source cutter-testing-framework
$ cd cutter-testing-framework-1.2.4/
$ uscan
$ uupdate -u ../cutter-testing-framework-1.2.5.tar.gz
$ cd ../cutter-testing-framework-1.2.5
$ debuild -S -us -uc -d
$ sudo cowbuilder --build  ../cutter-testing-framework_1.2.5-1.dsc
```

**Comment by kou** [comment]
Thanks. I got the following `cowbuilder` error. Do I need to run more commands?

```text
% sudo cowbuilder --build  ../cutter-testing-framework_1.2.5-1.dsc
I: Copying COW directory
I: forking: rm -rf /var/cache/pbuilder/build/cow.4514
I: forking: cp -al /var/cache/pbuilder/base.cow /var/cache/pbuilder/build/cow.4514
cp: cannot stat '/var/cache/pbuilder/base.cow': No such file or directory
E: Failed cowcopy.
```

**Comment by kenhys** [comment]
FYI: There is a possibility that similar failure is also happened on Ubuntu 16.04 (xenial) https://launchpad.net/~cutter-testing-framework/+archive/ubuntu/ppa/+build/9753681

**Comment by henrich** [comment]
Yes, before run cowbuilder, you should create base.cow via
```
$ sudo cowbuilder --create
```

**Comment by kou** [comment]
Thanks but still there is still a problem...:

```text
% sudo cowbuilder --build  ../cutter-testing-framework_1.2.5-1.dsc
...
cp: cannot stat '/tmp/cutter-testing-framework_1.2.5.orig.tar.gz': Permission denied
...
```

Full log:

```text
% sudo cowbuilder --build  ../cutter-testing-framework_1.2.5-1.dsc
I: Copying COW directory
I: forking: rm -rf /var/cache/pbuilder/build/cow.6323
I: forking: cp -al /var/cache/pbuilder/base.cow /var/cache/pbuilder/build/cow.6323
I: unlink for ilistfile /var/cache/pbuilder/build/cow.6323/.ilist failed, it didn't exist?
I: forking: chroot /var/cache/pbuilder/build/cow.6323 cowdancer-ilistcreate /.ilist 'find . -xdev -path ./home -prune -o \( \( -type l -o -type f \) -a -links +1 -print0 \) | xargs -0 stat --format '%d %i ''
I: Invoking pbuilder
I: forking: pbuilder build --buildplace /var/cache/pbuilder/build/cow.6323 --buildresult /var/cache/pbuilder/result/ --no-targz --internal-chrootexec 'chroot /var/cache/pbuilder/build/cow.6323 cow-shell' /tmp/cutter-testing-framework_1.2.5-1.dsc
W: /root/.pbuilderrc does not exist
I: Running in no-targz mode
I: using fakeroot in build.
I: pbuilder: network access will be disabled during build
I: Current time: Sat Apr  8 11:15:49 JST 2017
I: pbuilder-time-stamp: 1491617749
I: copying local configuration
I: mounting /proc filesystem
I: mounting /sys filesystem
I: creating /{dev,run}/shm
I: mounting /dev/pts filesystem
I: redirecting /dev/ptmx to /dev/pts/ptmx
I: policy-rc.d already exists
I: Obtaining the cached apt archive contents
ln: failed to create hard link '/var/cache/pbuilder/build/cow.6323/var/cache/apt/archives/libacl1_2.2.52-3+b1_amd64.deb': File exists
ln: failed to create hard link '/var/cache/pbuilder/build/cow.6323/var/cache/apt/archives/adduser_3.115_all.deb': File exists
ln: failed to create hard link '/var/cache/pbuilder/build/cow.6323/var/cache/apt/archives/apt_1.4_amd64.deb': File exists
I: Copying source file
I: copying [/tmp/cutter-testing-framework_1.2.5-1.dsc]
I: copying [/tmp/cutter-testing-framework_1.2.5.orig.tar.gz]
cp: cannot stat '/tmp/cutter-testing-framework_1.2.5.orig.tar.gz': Permission denied
I: unmounting dev/ptmx filesystem
I: unmounting dev/pts filesystem
I: unmounting dev/shm filesystem
I: unmounting proc filesystem
I: unmounting sys filesystem
I: Cleaning COW directory
I: forking: rm -rf /var/cache/pbuilder/build/cow.6323
```

**Comment by henrich** [comment]
Don't put files under /tmp, use ~/tmp or somewhere, instead.

**Comment by kou** [comment]
Thanks. I can reproduce it.
I want to run `cutter` under `gdb` to debug this. How do I do this...?

**Comment by henrich** [comment]
Do you want to run it inside of cowbuilder chroot?
Then, use pbuilder hook as /usr/share/doc/pbuilder/examples/C10shell 

1. Specify hookdir in /etc/pbuilderrc like "HOOKDIR=/home/henrich/pbuilder-hook"
1. mkdir /home/henrich/pbuilder-hook
1. cp /usr/share/doc/pbuilder/examples/C10shell /home/henrich/pbuilder-hook
1. run cowbuilder

**Comment by henrich** [comment]
fixed in #23
